eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/LazyDoctrineRepositoriesPass.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Compiler;
+
+use RuntimeException;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+final class LazyDoctrineRepositoriesPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        $nonLazyServices = [];
+        foreach ($container->getDefinitions() as $serviceId => $definition) {
+            if (!is_array($definition->getFactory())) {
+                continue;
+            }
+
+            $factory = $definition->getFactory();
+            $factoryServiceId = (string) $factory[0];
```

Comment:
This looks like the culprit here. This code is different in `ezplatform-kernel` somehow and it contains the right check against it being a `Reference` (which is stringable), not a `Definition`.